client/app/store.ts

```diff
@@ -0,0 +1,108 @@
+import disbursementReducer from './bundles/course/experience-points/disbursement/store';
+import forumsReducer from './bundles/course/forum/store';
+import groupsReducer from './bundles/course/group/store';
+import leaderboardReducer from './bundles/course/leaderboard/store';
+import learningMapReducer from './bundles/course/learning-map/store';
+import lessonPlanReducer from './bundles/course/lesson-plan/store';
+import levelsReducer from './bundles/course/level/store';
+import foldersReducer from './bundles/course/material/folders/store';
+import timelinesReducer from './bundles/course/reference-timelines/store';
+import statisticsReducer from './bundles/course/statistics/store';
+import surveysReducer from './bundles/course/survey/store';
+import userEmailSubscriptionsReducer from './bundles/course/user-email-subscriptions/store';
+import invitationsReducer from './bundles/course/user-invitations/store';
+import usersReducer from './bundles/course/users/store';
+import videosReducer from './bundles/course/video/store';
+import adminReducer from './bundles/system/admin/admin/store';
+import instanceAdminReducer from './bundles/system/admin/instance/instance/store';
+import globalUserReducer from './bundles/users/store';
+
+enableMapSet();
+
+const rootReducer = combineReducers({
+  disbursement: disbursementReducer,
+  submissions: submissionsReducer,
+  timelines: timelinesReducer,
+  global: combineReducers({
+    user: globalUserReducer,
+    announcements: globalAnnouncementReducer,
+  }),
+  achievements: achievementsReducer,
+  announcements: announcementsReducer,
+  skills: skillsReducer,
+  courses: coursesReducer,
+  comments: commentsReducer,
+  forums: forumsReducer,
+  groups: groupsReducer,
+  leaderboard: leaderboardReducer,
+  learningMap: learningMapReducer,
+  folders: foldersReducer,
+  videos: videosReducer,
+  levels: levelsReducer,
+  notificationPopup: notificationPopupReducer,
+  statistics: statisticsReducer,
+  users: usersReducer,
+  invitations: invitationsReducer,
+  enrolRequests: enrolRequestsReducer,
+  lessonPlanSettings: lessonPlanSettingsReducer,
+  notificationSettings: notificationSettingsReducer,
+  duplication: duplicationsReducer,
+  scribingQuestion: scribingQuestionReducer,
+  lessonPlan: lessonPlanReducer,
+  deleteConfirmation: deleteConfirmationReducer,
+  admin: adminReducer,
+  instanceAdmin: instanceAdminReducer,
+  surveys: surveysReducer,
+  userEmailSubscriptions: userEmailSubscriptionsReducer,
+  assessments: assessmentsReducer,
+});
+
+export const store = configureStore({
+  reducer: rootReducer,
+});
+
+export type AppState = ReturnType<typeof rootReducer>;
+
+export type AppDispatch = ThunkDispatch<
+  AppState,
+  Record<string, unknown>,
+  AnyAction
+>;
+
+export type Operation<R = void> = ThunkAction<
+  Promise<R>,
+  AppState,
+  Record<string, unknown>,
+  AnyAction
+>;
+
+export const dispatch = store.dispatch as AppDispatch;
+
+export const setUpTestStore = (
```

Comment:
If we move this to `TestApp`, means that we have to export `rootReducer` and have `configureStore` be imported by someone else. As much as possible, I prefer if we keep Redux store creation methods in here.

I'll rename `setUpTestStore` to `setUpStoreWithState` instead since it is nothing specific to tests. It just _happens_ to be used by tests.

`dispatch` above is actually also never meant for anyone to use. In fact, it is only used by tests. But I keep it there in case we need to dispatch from somewhere not in a component.